src/main/java/net/fortuna/ical4j/model/Recur.java

```diff
@@ -923,13 +929,62 @@ private void validateFrequency() {
         } else if (Frequency.MONTHLY.equals(getFrequency())) {
             calIncField = ChronoUnit.MONTHS;
         } else if (Frequency.YEARLY.equals(getFrequency())) {
-            calIncField = ChronoUnit.YEARS;
+            if (getWeekNoList().isEmpty()) {
+                calIncField = ChronoUnit.YEARS;
+            } else {
+                calIncField = weekBasedYears(WeekDay.getDayOfWeek(getWeekStartDay()));
```

Comment:
This was a weird rabbit hole: we want the `TemporalUnit` for  'week-based years' here (i.e. move to the date in the next year that has the same 'week of year' and 'day of week'), but the Java standard library doesn't appear to have an implementation that honors alternative ways of determining 'week 1': `WeekFields.WEEK_BASED_YEARS` exists but is just an alias for `IsoFields.WEEK_BASED_YEARS`.

LMK if you want me to move that helper to its own class.

Reply:
Thanks for this, much appreciated. 

Yes, so many rabbit holes when working with time and timezones. I think Recur and ZoneRulesBuilder might be the most complex classes in the library. ;-)